C/include/c4ReplicatorTypes.h

```diff
@@ -184,10 +188,32 @@ C4API_BEGIN_DECLS
 #endif // COUCHBASE_ENTERPRISE
 
 
+    typedef struct C4ReplicationCollection {
+        C4CollectionSpec collection;
+
+        C4ReplicatorMode                    push;              ///< Push mode (from db to remote/other db)
+        C4ReplicatorMode                    pull;              ///< Pull mode (from db to remote/other db).
```

Comment:
I understand that push and pull are per replication not collection.

Reply:
Theoretically, collections can be push/pull independently, but if we spec it that the same replicator shall have the same replicator mode, then we man move it out of collection area.

Reply:
FYI : The CBL API Spec have it the per replication.

Reply:
I can keep it collection specific now. Platforms just need to be little verbose to specify individually.

Reply:
Right now that doesn't support by the API. The replicationType is per replicator.

Reply:
LiteCore can make it supported, and we can revise the CBL API. Even we don't support now, support it later will not be breaking change.

Reply:
Okay, that's fine. I can just simply copy it multiple places.

Reply:
As Platforms convert CBLReplicatorConfig to C4ReplicatorConfig, they may assign the same push/pull mode to all collection area, cannot they?

Reply:
The platform can do that.